### DISCUSSION TITLE
Session replay

### DISCUSSION THREAD

**Discussion by kevinvugts** [stayallive/wp-sentry#145] [Q&A] [answered]
Are you planning to add support for the newly released Session Replay feature of Sentry?
If so, is there any ETA for this?

**Comment by stayallive** [comment]
Keep in mind that this is currently a beta feature. So I am not planning or looking into it at this moment. 

That said, I peeked at: https://docs.sentry.io/platforms/javascript/session-replay/. And it looked like the Browser bundle is all you need but looking here: https://docs.sentry.io/platforms/javascript/install/cdn/ it seems it requires another bundle to be loaded. 

I'm not sure how many bundles I want to keep adding so I might need to think a little about allowing these to be loaded from the CDN instead of bundling them. 

But the bundle seems to be the only thing needed. Setting options and adding integrations you can already do today. 

Not sure anything else is needed?

**Reply by aaronwjm360** [reply]
@stayallive I also am interested in this functionality. It looks like it requires @sentry/browser, which I think you already include, with the current configuration:

```
import * as Sentry from "@sentry/browser";

Sentry.init({
  dsn: "https://0c324789f2434acbb82392c21b0551c5@o4504808822472704.ingest.sentry.io/4504808840364032",

  // This sets the sample rate to be 10%. You may want this to be 100% while
  // in development and sample at a lower rate in production
  replaysSessionSampleRate: 0.1,
  // If the entire session is not sampled, use the below sample rate to sample
  // sessions when an error occurs.
  replaysOnErrorSampleRate: 1.0,

  integrations: [new Sentry.Replay()],
});
```

how would I enable this integration using your plugin? I think its just a case of adding something into the integrations?

**Reply by stayallive** [reply]
We currently ship the CDN bundles with this plugin and don't use the NPM packages, it seems like simply including the browser bundle is not enough, it would also require the replay bundle or added scripts (same goes for tracing which we already implement). Requires some more research and added coded than simply enabling it in the configuration unfortunately.

**Reply by aaronwjm360** [reply]
@stayallive no worries, thanks for your quick and prompt response and hopefully maybe see this feature in the future :)

**Reply by pawelkmpt** [reply]
Besides enqueuing the additional bundle initialization of the integration is needed. Basic solution [docs](https://docs.sentry.io/platforms/javascript/session-replay/#set-up) are are suggesting:

```js
Sentry.init({
  dsn: "https://examplePublicKey@o0.ingest.sentry.io/0",

  // This sets the sample rate to be 10%. You may want this to be 100% while
  // in development and sample at a lower rate in production
  replaysSessionSampleRate: 0.1,

  // If the entire session is not sampled, use the below sample rate to sample
  // sessions when an error occurs.
  replaysOnErrorSampleRate: 1.0,

  integrations: [
    new Sentry.Replay({
      // Additional SDK configuration goes in here, for example:
      maskAllText: true,
      blockAllMedia: true,
    }),
  ],
});
```

but I didn't manage to do it by using `wp_sentry_public_options` filter as it treats all things passed as string, not JS object.

[Alternative](https://docs.sentry.io/platforms/javascript/session-replay/configuration/#start-and-stop-recording):

```js
const replay = new Sentry.Replay();

// If you want to defer loading Replay
Sentry.getCurrentHub()
  .getClient()
  .addIntegration(replay);

// sometime later
await replay.flush(); // wait for any pending recording data to be sent
replay.stop();

replay.start();
```

I used second approach but I'm not sure if I did it correctly as I have just a few records. I'm waiting on support response right now.

**Comment by stayallive** [comment, marked as answer]
Session Replays SDK has been added: https://github.com/stayallive/wp-sentry/releases/tag/v6.10.0

You can start by setting the following constants:

```php
// https://docs.sentry.io/platforms/javascript/session-replay/configuration/#general-integration-configuration
define( 'WP_SENTRY_BROWSER_REPLAYS_SESSION_SAMPLE_RATE', 0.1 ); // replaysSessionSampleRate
define( 'WP_SENTRY_BROWSER_REPLAYS_ON_ERROR_SAMPLE_RATE', 1.0 ); // replaysOnErrorSampleRate
```
